samples/augment-reality-to-show-hidden-infrastructure/src/main/java/com/esri/arcgismaps/sample/augmentrealitytoshowhiddeninfrastructure/components/MapViewModel.kt

```diff
@@ -0,0 +1,127 @@
+/* Copyright 2025 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.augmentrealitytoshowhiddeninfrastructure.components
+
+import android.app.Application
+import androidx.compose.runtime.MutableFloatState
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableFloatStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.Color
+import com.arcgismaps.geometry.GeometryType
+import com.arcgismaps.geometry.Polyline
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.symbology.SimpleLineSymbol
+import com.arcgismaps.mapping.symbology.SimpleLineSymbolStyle
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.mapping.view.LocationDisplay
+import com.arcgismaps.mapping.view.geometryeditor.GeometryEditor
+import kotlinx.coroutines.launch
+
+class MapViewModel(app: Application) : AndroidViewModel(app) {
+
+    val arcGISMap = ArcGISMap(BasemapStyle.ArcGISImagery)
+    val graphicsOverlay = GraphicsOverlay()
+    val geometryEditor = GeometryEditor()
+
+    private val _statusText = mutableStateOf("Tap on map or use current location to create start point")
+    val statusText: MutableState<String> = _statusText
```

Comment:
Could use `by mutableStateOf` to help avoid using `.value` on every status update here: 
```suggestion
    private val _statusText by mutableStateOf("Tap on map or use current location to create start point")
    val statusText get() = _statusText
```